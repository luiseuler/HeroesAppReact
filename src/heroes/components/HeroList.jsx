import { getHeroesByPublisher } from "../helpers";
import { HeroCard } from "./HeroCard";

export const HeroList = ({ publisher }) => {
    const heroes = getHeroesByPublisher(publisher);

    return (
        <diiv className="row row-cols-1 row-cols-md-3 g-3">
            {
                heroes.map(hero => (<HeroCard key={hero.superhero} {...hero} />))
            }
        </diiv>
    );
}
