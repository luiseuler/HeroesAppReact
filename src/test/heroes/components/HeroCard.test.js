import { render, screen } from "@testing-library/react";
import { HeroCard } from "../../../heroes/components/";
import { MemoryRouter } from "react-router-dom";

describe('Pruebas en <HeroCard />', () => {
    let hero;

    beforeEach(() => {
        hero = {
            id: 'dc-batman',
            superhero: 'Batman',
            publisher: 'DC Comics',
            alter_ego: 'Bruce Wayne',
            first_appearance: 'Detective Comics #27',
            characters: 'Bruce Wayne'
        }
    });

    test('debe de hacer match con el snapshot', () => {
        const { container } = render(
            <MemoryRouter>
                <HeroCard {...hero} />
            </MemoryRouter>
        );

        expect(container).toMatchSnapshot();
    });

    test('si alter_ego y characters son iguales no debe de mostrar los characters', () => {
        render(
            <MemoryRouter>
                <HeroCard {...hero} />
            </MemoryRouter>
        );

        const charactersElement = screen.queryByLabelText('characters');
        expect(charactersElement).toBeNull();
    });

    test('si alter_ego y characters son diferentes se debe de mostrar los characters', () => {
        hero.characters = 'some another';

        render(
            <MemoryRouter>
                <HeroCard {...hero} />
            </MemoryRouter>
        );

        const charactersElement = screen.queryByLabelText('characters');
        expect(charactersElement).toBeTruthy();
    });

    test('se debe de construir la url de la imagen basado en id', () => {
        render(
            <MemoryRouter>
                <HeroCard {...hero} />
            </MemoryRouter>
        );

        const imgElement = screen.getByRole('img');

        expect(imgElement.src).toContain(`${hero.id}.jpg`);
    });
});