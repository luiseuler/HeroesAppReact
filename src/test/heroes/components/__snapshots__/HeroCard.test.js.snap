// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HeroCard /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="col animate__animated animate__fadeIn"
  >
    <div
      class="card"
    >
      <div
        class="row no-gutters"
      >
        <div
          class="col-4"
        >
          <img
            alt="Batman"
            class="card-img"
            src="/heroes/dc-batman.jpg"
          />
        </div>
        <div
          class="col-8"
        >
          <div
            class="card-body"
          >
            <h5
              class="card-title"
            >
              Batman
            </h5>
            <p
              class="card-text"
            >
              Bruce Wayne
            </p>
            <p
              class="card-text"
            >
              <small
                class="text-muted"
              >
                Detective Comics #27
              </small>
            </p>
            <a
              href="/hero/dc-batman"
            >
              MÃ¡s...
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
